modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/web/internal/info/item/provider/test/ObjectEntryInfoItemFormProviderTest.java

```diff
@@ -450,6 +461,75 @@ private void _testGetInfoFormWithPicklistObjectField() throws Exception {
 		}
 	}
 
+	@TestInfo("LPD-66678")
+	private void _testGetInfoFormWithRootObjectDefinition() throws Exception {
+		Tree tree = TreeTestUtil.createObjectDefinitionTree(
+			_objectDefinitionLocalService, _objectRelationshipLocalService,
+			false,
+			LinkedHashMapBuilder.put(
+				"A", new String[] {"AA"}
+			).put(
+				"AA", new String[0]
+			).build());
+
+		Node rootNode = tree.getRootNode();
+
+		ObjectDefinition rootNodeObjectDefinition =
+			_objectDefinitionLocalService.getObjectDefinition(
+				rootNode.getPrimaryKey());
+
+		_objectDefinitionLocalService.publishCustomObjectDefinition(
+			TestPropsValues.getUserId(),
+			rootNodeObjectDefinition.getObjectDefinitionId());
+
+		InfoForm rootNodeInfoForm = _getInfoForm(rootNodeObjectDefinition);
+
+		_assertInfoField("able", rootNodeInfoForm);
+
+		List<Node> childNodes = rootNode.getChildNodes();
+
+		Node childNode = childNodes.get(0);
+
+		ObjectDefinition childNodeObjectDefinition =
+			_objectDefinitionLocalService.getObjectDefinition(
+				childNode.getPrimaryKey());
+
+		_objectDefinitionLocalService.publishCustomObjectDefinition(
+			TestPropsValues.getUserId(),
+			childNodeObjectDefinition.getObjectDefinitionId());
+
+		InfoForm childNodeInfoForm = _getInfoForm(childNodeObjectDefinition);
+
+		_assertInfoField("able", childNodeInfoForm);
+
+		InfoFieldSet infoFieldSet =
+			(InfoFieldSet)rootNodeInfoForm.getInfoFieldSetEntry(
+				rootNodeObjectDefinition.getName());
+
+		List<ObjectRelationship> objectRelationships =
+			_objectRelationshipLocalService.getObjectRelationships(
+				rootNodeObjectDefinition.getObjectDefinitionId());
+
+		_objectRelationship = objectRelationships.get(0);
+
+		Assert.assertNull(
+			infoFieldSet.getInfoFieldSetEntry(_objectRelationship.getName()));
+
+		infoFieldSet = (InfoFieldSet)childNodeInfoForm.getInfoFieldSetEntry(
+			childNodeObjectDefinition.getName());
```

Comment:
Previously, you declared two variable: `rootNodeInfoForm` and `childNodeInfoForm`.
Here, you decided to reuse  the variable `infoFieldSet` and used no prefix. I believe that you should decide: reuse variables `InfoForm` and `infoFieldSet` OR add prefix on both of them.

I think it make sense to add prefix on both. So, can you rename it to:
`rootNodeInfoFieldSet` and `childNodeInfoFieldSet`